specification/frontdoor/resource-manager/Microsoft.Network/stable/2019-03-01/webapplicationfirewall.json

```diff
@@ -0,0 +1,813 @@
+        "tags": [
+          "WebApplicationFirewallManagedRuleSets"
+        ],
+        "description": "Lists all available managed rule sets.",
+        "operationId": "ManagedRuleSets_List",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The operation returns a list of all available web application firewall managed rule sets.",
+            "schema": {
+              "$ref": "#/definitions/ManagedRuleSetDefinitionList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List Policies in a Resource Group": {
+            "$ref": "./examples/WafListManagedRuleSets.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "WebApplicationFirewallPolicy": {
+      "description": "Defines web application firewall policy.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties of the web application firewall policy.",
+          "$ref": "#/definitions/WebApplicationFirewallPolicyProperties"
+        },
+        "etag": {
+          "type": "string",
+          "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ]
+    },
+    "WebApplicationFirewallPolicyProperties": {
+      "description": "Defines web application firewall policy properties.",
+      "properties": {
+        "policySettings": {
+          "description": "Describes settings for the policy.",
+          "$ref": "#/definitions/PolicySettings"
+        },
+        "customRules": {
+          "description": "Describes custom rules inside the policy.",
+          "$ref": "#/definitions/CustomRuleList"
+        },
+        "managedRules": {
+          "description": "Describes managed rules inside the policy.",
+          "$ref": "#/definitions/ManagedRuleSetList"
+        },
+        "frontendEndpointLinks": {
+          "description": "Describes Frontend Endpoints associated with this Web Application Firewall policy.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/FrontendEndpointLink"
+          }
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "description": "Provisioning state of the policy.",
+          "$ref": "#/definitions/ProvisioningState"
```

Comment:
note: The type is changed, compare to preview version. It was a string.

Reply:
The enumerations are the same as what the API returns currently. I forgot this would be a breaking change for SDKs. I can revert this for now and include it in a future API version.

---
In reply to: [285736252](https://github.com/Azure/azure-rest-api-specs/pull/5918#discussion_r285736252) [](ancestors = 285736252)